### PR TITLE
feat: Add parser container_inspect

### PR DESCRIPTION
**Pull request by wushiqinlou**
Signed-off-by: jiazhang <jiazhang@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:
This PR is used to collect the output of "docker|podman inspect <container ID>" for new advisor rule.

### CODE REVIEW

**Comment by wushiqinlou** [comment]
@xiangce - Please review.

File: insights/specs/datasources/containers_inspect.py
Comment (xiangce) [inline insights/specs/datasources/containers_inspect.py:61]: I don't think it's correct to make this `_find` a generator, since it may miss the keys which is at the end of the `filters` but locate in the top of the `json`.

File: insights/specs/datasources/containers_inspect.py
Comment (xiangce) [reply insights/specs/datasources/containers_inspect.py:75]: A full version which fixed the `_find` generator:
```python
    def _get_kvs(key, _dict):
        ret = {}
        for k, v in _dict.items():
            if key == k:
                ret.update({k: v})
            elif isinstance(v, dict):
                ret.update(_get_kvs(key, v))
        return ret

    try:
        filters = get_filters(Specs.containers_inspect_vars)
        total_results = []
        if filters:
            for item in broker[LocalSpecs.containers_inspect_data_raw]:
                raw_data = json.loads(''.join(item.content))[0]
                engine, _, container_id = item.cmd.split(None)
                filter_result=dict(Id=container_id, engine=os.path.basename(engine))
                for filt in filters:
                    filter_result.update(_get_kvs(filt, raw_data))
                total_results.append(filter_result) if filter_result else None
```

File: insights/parsers/containers_inspect.py
Comment (xiangce) [inline insights/parsers/containers_inspect.py:36]: Since the usage of the filter to this parser is different from normal specs, please add an example here.

**Review by xiangce — CHANGES_REQUESTED** [review]
Besides the following comments, the "podman_inspect.py" should also be deprecated.

File: insights/specs/datasources/containers_inspect.py
Comment (xiangce) [inline insights/specs/datasources/containers_inspect.py:63]: The `v` could be a `list`.

**Comment by wushiqinlou** [comment]
@xiangce - I updated this PR according to your comments, please check.

**Comment by wushiqinlou** [comment]
New collected archive:

~~~
# cat insights_containers/insights_containers

[{"engine": "podman", "Config|Cmd": ["sleep", "100000"], "Image": "acf3e09a39c95d354539b6591298be0b0814f5d74e95e722863241192b9a079b", "Id": "81b8de0d16ce", "HostConfig|Privileged": false}, {"engine": "podman", "Config|Cmd": ["sleep", "100000"], "Image": "acf3e09a39c95d354539b6591298be0b0814f5d74e95e722863241192b9a079b", "Id": "327202b625d0", "HostConfig|Privileged": false}, {"engine": "docker", "Config|Cmd": ["sleep", "100000"], "Image": "acf3e09a39c95d354539b6591298be0b0814f5d74e95e722863241192b9a079b", "Id": "706c223d3f58", "HostConfig|Privileged": false}, {"engine": "docker", "Config|Cmd": ["sleep", "100000"], "Image": "acf3e09a39c95d354539b6591298be0b0814f5d74e95e722863241192b9a079b", "Id": "a7c6b05025f4", "HostConfig|Privileged": false}, {"engine": "docker", "Config|Cmd": ["sleep", "1000000"], "Image": "acf3e09a39c95d354539b6591298be0b0814f5d74e95e722863241192b9a079b", "Id": "c7efee959ea8", "HostConfig|Privileged": true}]
~~~

**Review by xiangce — APPROVED** [review]
Approved pending the approval of new specs

File: insights/specs/datasources/containers_inspect.py
Comment (psachin) [reply insights/specs/datasources/containers_inspect.py]: How about using `pathlib` instead which is better than `os.path..`  [PEP 428](https://peps.python.org/pep-0428/)?

File: insights/specs/datasources/container/containers_inspect.py
Comment (psachin) [inline insights/specs/datasources/container/containers_inspect.py:87]: Can you add tests for these lines?

Comment (wushiqinlou) [reply insights/specs/datasources/container/containers_inspect.py:87]: @psachin - Generally this condition is always True, I added this condition in case there is unexpected situation that `total_results` is `False`, I will add a comment to explain it.

Comment (xiangce) [reply insights/specs/datasources/container/containers_inspect.py:87]: Hm... the `False` branch of this line will never reach, since the `required` component `LocalSpecs.containers_inspect_data_raw` guaranteed the `loop` of L69 must be trapped at least once. Then the `total_results` must include at least one `filter_results`.

However, I think it's worth leaving this check here, as an `explicit` after assertion.

File: insights/specs/datasources/containers_inspect.py
Comment (xiangce) [reply insights/specs/datasources/containers_inspect.py]: I didn't see 'pathlib' is broad used in the core, for now, I believe we'd better keep `os.path`  as it works as expected.

**Pull request merged**
